4 tests fail when run in Koji or on EL 7, but not in the development environment

I found today that 4 tests are failing when the test suite is run in Koji (on any Fedora release), or when run on my EL 7 box. Here's an example:

```
$ python setup.py nosetests -sx
running nosetests
running egg_info
writing bodhi.egg-info/PKG-INFO
writing top-level names to bodhi.egg-info/top_level.txt
writing dependency_links to bodhi.egg-info/dependency_links.txt
reading manifest file 'bodhi.egg-info/SOURCES.txt'
reading manifest template 'BODHI_MANIFEST.in'
writing manifest file 'bodhi.egg-info/SOURCES.txt'
running build_ext
..........................................................................................F
======================================================================
FAIL: Assert correct operation when the repomd checksum matches immediately.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mock.py", line 1618, in _inner
    return f(*args, **kw)
  File "/usr/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/home/rbarlow/devel/bodhi/bodhi/tests/server/consumers/test_masher.py", line 1372, in test_checksum_match_immediately
    'http://example.com/pub/fedora/linux/updates/testing/17/x86_64/repodata.repomd.xml')
  File "/usr/lib/python2.7/site-packages/mock.py", line 846, in assert_called_once_with
    return self.assert_called_with(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/mock.py", line 835, in assert_called_with
    raise AssertionError(msg)
AssertionError: Expected call: urlopen('http://example.com/pub/fedora/linux/updates/testing/17/x86_64/repodata.repomd.xml')
Actual call: urlopen(u'http://example.com/pub/fedora/linux/updates/testing/17/aarch64/repodata.repomd.xml')
```

I'm not sure why I can't reproduce this on the development environment, but I do have a local EL 7 box that I can reproduce it with so that's nice.

--- a/bodhi/tests/server/consumers/test_masher.py
+++ b/bodhi/tests/server/consumers/test_masher.py
@@ -1368,8 +1368,16 @@ class TestMasherThread_wait_for_sync(MasherThreadBaseTestCase):
             mock.call(topic='mashtask.sync.done', msg={'repo': t.id, 'agent': 'bowlofeggs'},
                       force=True)]
         publish.assert_has_calls(expected_calls)
-        urlopen.assert_called_once_with(
-            'http://example.com/pub/fedora/linux/updates/testing/17/x86_64/repodata.repomd.xml')
+        self.assertEqual(urlopen.call_count, 1)
+        # Since os.listdir() isn't deterministic about the order of the items it returns, the test
+        # won't be deterministic about which of these URLs get called. However, either one of them
+        # would be correct so we will just assert that one of them is called.
+        expected_calls = [
+            mock.call('http://example.com/pub/fedora/linux/updates/testing/17/x86_64/'
+                      'repodata.repomd.xml'),
+            mock.call('http://example.com/pub/fedora/linux/updates/testing/17/aarch64/'
+                      'repodata.repomd.xml')]
+        self.assertTrue(urlopen.mock_calls[0] in expected_calls)
 
     @mock.patch.dict(
         'bodhi.server.consumers.masher.config',
@@ -1403,9 +1411,13 @@ class TestMasherThread_wait_for_sync(MasherThreadBaseTestCase):
             mock.call(topic='mashtask.sync.done', msg={'repo': t.id, 'agent': 'bowlofeggs'},
                       force=True)]
         publish.assert_has_calls(expected_calls)
+        # Since os.listdir() isn't deterministic about the order of the items it returns, the test
+        # won't be deterministic about which of arch URL gets used. However, either one of them
+        # would be correct so we will just assert that the one that is used is used correctly.
+        arch = 'x86_64' if 'x86_64' in urlopen.mock_calls[0][1][0] else 'aarch64'
         expected_calls = [
-            mock.call(
-                'http://example.com/pub/fedora/linux/updates/testing/17/x86_64/repodata.repomd.xml')
+            mock.call('http://example.com/pub/fedora/linux/updates/testing/17/'
+                      '{}/repodata.repomd.xml'.format(arch))
             for i in range(3)]
         urlopen.assert_has_calls(expected_calls)
         sleep.assert_has_calls([mock.call(200), mock.call(200)])
@@ -1444,9 +1456,13 @@ class TestMasherThread_wait_for_sync(MasherThreadBaseTestCase):
             mock.call(topic='mashtask.sync.done', msg={'repo': t.id, 'agent': 'bowlofeggs'},
                       force=True)]
         publish.assert_has_calls(expected_calls)
+        # Since os.listdir() isn't deterministic about the order of the items it returns, the test
+        # won't be deterministic about which of arch URL gets used. However, either one of them
+        # would be correct so we will just assert that the one that is used is used correctly.
+        arch = 'x86_64' if 'x86_64' in urlopen.mock_calls[0][1][0] else 'aarch64'
         expected_calls = [
-            mock.call(
-                'http://example.com/pub/fedora/linux/updates/testing/17/x86_64/repodata.repomd.xml')
+            mock.call('http://example.com/pub/fedora/linux/updates/testing/17/'
+                      '{}/repodata.repomd.xml'.format(arch))
             for i in range(2)]
         urlopen.assert_has_calls(expected_calls)
         t.log.exception.assert_called_once_with('Error fetching repomd.xml')
@@ -1542,9 +1558,13 @@ class TestMasherThread_wait_for_sync(MasherThreadBaseTestCase):
             mock.call(topic='mashtask.sync.done', msg={'repo': t.id, 'agent': 'bowlofeggs'},
                       force=True)]
         publish.assert_has_calls(expected_calls)
+        # Since os.listdir() isn't deterministic about the order of the items it returns, the test
+        # won't be deterministic about which of arch URL gets used. However, either one of them
+        # would be correct so we will just assert that the one that is used is used correctly.
+        arch = 'x86_64' if 'x86_64' in urlopen.mock_calls[0][1][0] else 'aarch64'
         expected_calls = [
-            mock.call(
-                'http://example.com/pub/fedora/linux/updates/testing/17/x86_64/repodata.repomd.xml')
+            mock.call('http://example.com/pub/fedora/linux/updates/testing/17/'
+                      '{}/repodata.repomd.xml'.format(arch))
             for i in range(2)]
         urlopen.assert_has_calls(expected_calls)
         t.log.exception.assert_called_once_with('Error fetching repomd.xml')

